lms/djangoapps/courseware/tests/test_middleware.py

```diff
@@ -26,32 +25,24 @@ def setUpClass(cls):
     def setUp(self):
         super(CoursewareMiddlewareTestCase, self).setUp()
 
-    def check_user_not_enrolled_redirect(self):
-        """A UserNotEnrolled exception should trigger a redirect"""
-        request = RequestFactory().get("dummy_url")
-        response = RedirectUnenrolledMiddleware().process_exception(
-            request, courses.UserNotEnrolled(self.course.id)
-        )
-        self.assertEqual(response.status_code, 302)
-        # make sure we redirect to the course about page
-        expected_url = reverse(
-            "about_course", args=[self.course.id.to_deprecated_string()]
```

Comment:
Good question. To my mind, this is a unit test of the middleware, and that no longer has any special knowledge of course enrollment (that's why I renamed it too). There are tests of the redirects, e.g. ``test_redirection_unenrolled``.